rotkehlchen/chain/evm/decoding/thegraph/decoder.py

```diff
@@ -62,16 +58,56 @@ def __init__(
             base_tools=base_tools,
             msg_aggregator=msg_aggregator,
         )
+        CustomizableDateMixin.__init__(self, base_tools.database)
         self.token = native_asset.resolve_to_evm_token()
         self.staking_contract = staking_contract
 
     def _decode_delegator_staking(self, context: DecoderContext) -> EvmDecodingOutput:
-        if context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED:
-            return self._decode_stake_delegated(context)
-        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
-            return self._decode_stake_locked(context)
-        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
-            return self._decode_stake_withdrawn(context)
+        """Decode delegator staking events for The Graph protocol.
+        Handles events from both pre and post Horizon upgrade contracts.
+        """
+        if (topic := context.tx_log.topics[0]) == TOPIC_STAKE_DELEGATED:
+            return self._decode_stake_delegated(
+                context=context,
+                delegator=bytes_to_address(context.tx_log.topics[2]),
+                verifier=None,
+            )
+        elif topic == TOPIC_STAKE_DELEGATED_HORIZON:
+            return self._decode_stake_delegated(
+                context=context,
+                delegator=bytes_to_address(context.tx_log.topics[3]),
+                verifier=bytes_to_address(context.tx_log.topics[2]),
+            )
+        elif topic == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(
+                context=context,
+                delegator=bytes_to_address(context.tx_log.topics[2]),
+                lock_duration_msg=f' Lock expires at epoch {int.from_bytes(context.tx_log.data[64:128])}',  # noqa: E501
```

Comment:
The space prefix should be part of `_decode_stake_locked`

```suggestion
                lock_duration_msg=f'Lock expires at epoch {int.from_bytes(context.tx_log.data[64:128])}',  # noqa: E501
```